main.tf

```diff
@@ -394,14 +394,13 @@ resource "local_file" "eks_config_map_aws_auth" {
 resource "null_resource" "aws_iam_authenticator" {
   provisioner "local-exec" {
     command = <<EOF
-if [ \"$(uname)\" == \"Darwin\" ]; \
-  then curl -o aws-iam-authenticator https://amazon-eks.s3-us-west-2.amazonaws.com/1.13.7/2019-06-11/bin/darwin/amd64/aws-iam-authenticator; \
-elif [ \"$(expr substr $(uname -s) 1 5)\" == \"Linux\" ]; \
```

Comment:
ya - there was no reason to use expr here - a simple `elif [ "$(uname -s)" == "Linux" ]` would have sufficed

Reply:
but what you have with `else` and removing conditional also works